### DISCUSSION TITLE
Is there any way to run ONNX with JAVA 7?

### DISCUSSION THREAD

**Discussion by johanneshk** [microsoft/onnxruntime#8558] [Other Q&A] [answered]
Hello,
currently the minimum supported JAVA version is 8. We want to deploy a model into an environment where JAVA 7 is installed (and we cannot change that). Is there any way to run with JAVA 7, e.g. if a previous version of ONNX supported this JAVA version?

We want to export a sklearn model and then run it with the JAVA runtime.

Thanks,
 Johannes

**Comment by Craigacp** [comment, marked as answer]
The ONNX Runtime Java API has always required Java 8, it was added in 2019, 4 years after Java 7 stopped receiving public updates. You might be able to get rid of the small number of Optional uses and recompile it for Java 7, but if you hit any issues you're on your own.
